<launch>
    <param name="use_sim_time" value="false"/>
    <param name="do_acc_boards" type="bool" value="false"/>
    <param name="end_process" type="bool" value="false"/>
    <param name="pause_processd" type="bool" value="false"/>
    <param name="cam_paused" type="bool" value="false"/>
    <param name="livox_paused" type="bool" value="false"/>
    
    <!-- the number of frames accumulated to do the calibration, default:60 -->
    <arg name="ns_l1" default="laser1"/>
    <arg name="ns_l2" default="laser2"/>
    <arg name="center_laser1_tp" default="/$(arg ns_l1)/centers_cloud"/>
    <arg name="center_laser2_tp" default="/$(arg ns_l2)/centers_cloud"/>
    <arg name="featureFileName" default="features_info_$(arg ns_l1)_to_$(arg ns_l2)"/>
    <arg name="max_frame" default="60"/>


    <param name="max_frame" type="int" value="$(arg max_frame)"/>
    
    <!-- Main Node -->
    <node pkg="lvt2calib" type="pattern_collection_ll" name="pattern_collection_ll" output="screen">
    
        <remap from="~cloud_laser1" to="$(arg center_laser1_tp)"/>
        <remap from="~cloud_laser2" to="$(arg center_laser2_tp)"/>
        
        <param name="ns_l1" value="$(arg ns_l1)"/>
        <param name="ns_l2" value="$(arg ns_l2)"/>
        <!-- use centroid of laser patterns? default:false -->
        <param name="useCentroid_laser" value="false" />

        <!-- Save file? -->        
        <param name="save_final_data" value="true"/>
        <!-- MUST TRUE!!! -->

        <!-- File Path -->
        <param name="result_dir_" type="string" value="$(find lvt2calib)/data/pattern_collection_result/"/>
        <param name="feature_file_name" type="string" value="$(arg featureFileName)"/>
    
    </node>
</launch>